sorting/radix_sort_2.c

```diff
@@ -1,71 +1,213 @@
 // sorting of array list using Radix sort
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #define range 10  // Range for integers is 10 as digits range from 0-9
 
-// Utility function to get the maximum value in ar[]
-int MAX(int *ar, int size)
+// Utility function to get the maximum value in arr[]
+int MAX(int *arr, int size)
 {
-    int i, max = ar[0];
+    int i, max = arr[0];
     for (i = 0; i < size; i++)
     {
-        if (ar[i] > max)
-            max = ar[i];
+        if (arr[i] > max)
+            max = arr[i];
     }
     return max;
 }
 
+// Utility function to get the minimum value in arr[]
+int MIN(int *arr, int size)
+{
+    int i, min = arr[0];
+    for (i = 0; i < size; i++)
+    {
+        if (arr[i] < min)
+            min = arr[i];
+    }
+    return min;
+}
+
 // Counting sort according to the digit represented by place
-void countSort(int *arr, int n, int place)
+void countSort(int *arr3, int pos, int place)
 {
     int i, freq[range] = {0};
-    int *output = (int *)malloc(n * sizeof(int));
+    int *output = (int *)malloc(pos * sizeof(int));
 
     // Store count of occurrences in freq[]
-    for (i = 0; i < n; i++) freq[(arr[i] / place) % range]++;
+    for (i = 0; i < pos; i++) freq[(arr3[i] / place) % range]++;
 
     // Change freq[i] so that it contains the actual position of the digit in
     // output[]
     for (i = 1; i < range; i++) freq[i] += freq[i - 1];
 
     // Build the output array
-    for (i = n - 1; i >= 0; i--)
+    for (i = pos - 1; i >= 0; i--)
     {
-        output[freq[(arr[i] / place) % range] - 1] = arr[i];
-        freq[(arr[i] / place) % range]--;
+        output[freq[(arr3[i] / place) % range] - 1] = arr3[i];
+        freq[(arr3[i] / place) % range]--;
     }
 
     // Copy the output array to arr[], so it contains numbers according to the
     // current digit
-    for (i = 0; i < n; i++) arr[i] = output[i];
+    for (i = 0; i < pos; i++) arr3[i] = output[i];
     free(output);
 }
 
+// Reverse counting sort according to the digit represented by place
+void reverseSort(int *arr2, int neg, int place)

```

Comment:
since arr2 is a parameter i think its fine to just use arr
same with all the others